src/externalized/DefaultContentManager.cc

```diff
@@ -426,13 +420,13 @@ SGPFile* DefaultContentManager::openGameResForReading(const char* filename) cons
 				// failed to open in the data dir
 				// let's try libraries
 
-				LibraryFile* libFile = LibraryFile_open(m_libraryDB, filename);
+				RustPointer<LibraryFile> libFile(LibraryFile_open(m_libraryDB.get(), filename));
 				if (libFile)
 				{
 					SLOGD("Opened file (from library ): %s", filename);
 					SGPFile *file = MALLOCZ(SGPFile);
 					file->flags = SGPFILE_NONE;
-					file->u.lib = libFile;
+					file->u.lib = libFile.release();
```

Comment:
Wouldn't it make sense to store the `RustPointer<_>` in `file->u.lib` and reset when `SGPFile` is destroyed?

Reply:
I don't feel that SGPFile is safe to use like that. I can't pinpoint the reason why, but when the feeling is this strong it usually means I got "burned" by something similiar in the past.